OsmAnd/src/net/osmand/plus/views/AutoZoomBySpeedUtils.java

```diff
@@ -0,0 +1,253 @@
+package net.osmand.plus.views;
+
+import com.github.mikephil.charting.charts.LineChart;
+import com.github.mikephil.charting.components.YAxis;
+import com.github.mikephil.charting.data.Entry;
+
+import net.osmand.core.android.MapRendererView;
+import net.osmand.data.RotatedTileBox;
+import net.osmand.gpx.GPXTrackAnalysis;
+import net.osmand.gpx.GPXTrackAnalysis.TrackPointsAnalyser;
+import net.osmand.gpx.PointAttributes;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.charts.ChartUtils;
+import net.osmand.plus.charts.GPXDataSetAxisType;
+import net.osmand.plus.charts.GPXDataSetType;
+import net.osmand.plus.charts.OrderedLineDataSet;
+import net.osmand.plus.settings.backend.OsmandSettings;
+import net.osmand.plus.settings.enums.AutoZoomMap;
+import net.osmand.plus.utils.ColorUtilities;
+import net.osmand.plus.utils.OsmAndFormatter;
+import net.osmand.plus.views.Zoom.ComplexZoom;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class AutoZoomBySpeedUtils {
+
+	public static final float ZOOM_PER_SECOND = 0.1f;
+	public static final int MIN_ZOOM_DURATION_MILLIS = 1500;
+	public static final long FIXED_ZOOM_DURATION_MILLIS = 1000;
+	private static final int SHOW_DRIVING_SECONDS = 60;
+
+	@Nullable
+	public static ComplexZoom calculateAutoZoomBySpeed(@NonNull OsmandApplication app, @NonNull RotatedTileBox tb, float speed) {
+		OsmandSettings settings = app.getSettings();
+
+		float zoomProximityCoeff = settings.AUTO_ZOOM_MAP_SCALE.get().coefficient;
+		float zoomDelta = defineZoomFromSpeed(tb, speed, zoomProximityCoeff);
+		if (Math.abs(zoomDelta) < 0.5) {
+			return null;
+		}
+
+		if (zoomDelta >= 2) {
```

Comment:
are zoom delta correct evaluations? what it means

Reply:
It's moved code for V1